third_party/amd/lib/TritonAMDGPUToLLVM/ElementwiseOpToLLVM.cpp

```diff
@@ -1078,20 +1197,22 @@ struct FpToFpOpConversion
     }
 
     // numElements = 4 for conversions:
-    // ocp bf8->fp16, ocp bf8->bf16, ocp bf8->fp32 on non-CDNA4
-    // fp16->ocp bf8, bf16->ocp bf8, fp32->ocp bf8(RTNE) on non-CDNA4
-    // fp32->ocp bf8(RTZ)
+    // ocp bf8->bf16, or
+    // ocp bf8->fp32/fp16 on non-CDNA4, or
+    // fp32/bf16/fp16->ocp bf8 on non-CDNA4
+    // fp32/bf16/fp16->ocp bf8 (RTZ) on CDNA4
     size_t numElements = 2;
-    if ((llvm::isa<Float8E5M2Type>(srcElementType) &&
-         !llvm::isa<Float32Type>(dstElementType)) ||
-        (llvm::isa<Float8E5M2Type>(srcElementType) &&
-         isaFamily != AMD::ISAFamily::CDNA4) ||
-        (!llvm::isa<Float32Type>(srcElementType) &&
-         llvm::isa<Float8E5M2Type>(dstElementType)) ||
-        (llvm::isa<Float32Type>(srcElementType) &&
-         llvm::isa<Float8E5M2Type>(dstElementType) &&
-         (roundingMode != RoundingMode::RTNE ||
-          isaFamily != AMD::ISAFamily::CDNA4))) {
+    if (llvm::isa<Float8E5M2Type>(srcElementType) &&
```

Comment:
Please use `(..)` to wrap `&&` cases like before to avoid clang warnings, which was addressed in https://github.com/triton-lang/triton/commit/4e5e465da6383da461dad21e723af04a81a63fe1.